api/lib/opentelemetry/resources/resource.rb

```diff
@@ -9,18 +9,41 @@ module Resources
     # Resource represents a resource, which captures identifying information about the entities
     # for which telemetry (metrics or traces) is reported.
     class Resource
+      # Returns the labels for this {Resource}
+      #
+      # @return [Hash<String, String>]
       attr_reader :labels
 
-      def initialize(**kvs)
+      # Returns a newly created {Resource} with the specified labels
+      #
+      # @param [Hash<String, String>] kvs Hash of key-value pairs to be used
+      #   as labels for this resource
+      # @return [Resource]
+      def initialize(kvs = {})
         # TODO: how defensive should we be here?
-        @labels = kvs.to_h { |k, v| [k.to_s.clone.freeze, v.to_s.clone.freeze] }.freeze
+        @labels = kvs.each_with_object({}) do |(k, v), memo|
+          memo[-k] = -v
+        end
```

Comment:
Should `.freeze` `@labels` after initialization. The [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-resources.md#retrieve-labels) requires `labels` to be exposed as

> a read only collection of labels